compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -836,6 +836,180 @@ fn should_encode_mir(tcx: TyCtxt<'_>, def_id: LocalDefId) -> (bool, bool) {
     }
 }
 
+fn should_encode_variances(def_kind: DefKind) -> bool {
+    match def_kind {
+        DefKind::Struct
+        | DefKind::Union
+        | DefKind::Enum
+        | DefKind::Variant
+        | DefKind::Fn
+        | DefKind::Ctor(..)
+        | DefKind::AssocFn => true,
+        DefKind::Mod
+        | DefKind::Field
+        | DefKind::AssocTy
+        | DefKind::AssocConst
+        | DefKind::TyParam
+        | DefKind::ConstParam
+        | DefKind::Static
+        | DefKind::Const
+        | DefKind::ForeignMod
+        | DefKind::TyAlias
+        | DefKind::OpaqueTy
+        | DefKind::Impl
+        | DefKind::Trait
+        | DefKind::TraitAlias
+        | DefKind::Macro(..)
+        | DefKind::ForeignTy
+        | DefKind::Use
+        | DefKind::LifetimeParam
+        | DefKind::AnonConst
+        | DefKind::GlobalAsm
+        | DefKind::Closure
+        | DefKind::Generator
+        | DefKind::ExternCrate => false,
```

Comment:
```suggestion
        DefKind::AssocFn
        | DefKind::Ctor(..)
        | DefKind::Enum
        | DefKind::Fn
        | DefKind::Struct
        | DefKind::Union
        | DefKind::Variant => true,
        DefKind::AnonConst
        | DefKind::AssocConst
        | DefKind::AssocTy
        | DefKind::Closure
        | DefKind::Const
        | DefKind::ConstParam
        | DefKind::ExternCrate
        | DefKind::Field
        | DefKind::ForeignMod
        | DefKind::ForeignTy
        | DefKind::Generator
        | DefKind::GlobalAsm
        | DefKind::Impl
        | DefKind::LifetimeParam
        | DefKind::Macro(..)
        | DefKind::Mod
        | DefKind::OpaqueTy
        | DefKind::Static
        | DefKind::Trait
        | DefKind::TraitAlias
        | DefKind::TyAlias
        | DefKind::TyParam
        | DefKind::Use => false,
```
sorted

Reply:
I don't think we should sort these differently from how they are defined in the enum.

Reply:
curious is that a convention? I have *very slight* attachment to the proposed change, though I guess the enum could be sorted also?

Reply:
The convention is: no order except for the one chosen by the author. While I think we could change things here, as long as rustfmt does not allow doing it automatically, I don't think we should do it. This is the stance we have on formatting/style nits in general, unless there is zero contention on the change. Since there is nonzero contention (namely mine 😝 ), we should not do this without rustfmt or other tooling support.

I'm going to go ahead and mark your changes as resolved, so the PR author doesn't have to look through the comments, but please don't hesitate to pursue a rustfmt rfc, as I think this is something that is very reasonable for large enums like this.

Reply:
very reasonable thanks